### PR TITLE
Python Wrapper for NINJA

### PR DESCRIPTION
**Pull request by bhillmann**
I ran into a bug where the Python wrapper doesn't properly handle file paths in Mac OSX. Started cleaning up the file so it should become more universal. Not ready yet for merge.

Note to view diff:
### Ignore Whitespace

Adding `?w=1` to any diff URL will remove any changes only in whitespace, enabling you to see only that code that has changed.

### CODE REVIEW

File: ninja.py [hunk 1]
```diff
@@ -107,11 +107,11 @@ def check_args(args, p, check_fasta):
     if args['input'] is None:
         p.print_help()
         sys.exit('\nPlease include an input sequences file in fasta format.')
-    
+
     # Checks if input sequences fasta is correctly formatted. Writes correct one if not
     elif check_fasta:
         fileName = args['input']
-        if not check_fasta(open(fileName)):
+        if not check_fasta2(open(fileName)):
```
Comment (bhillmann) [inline ninja.py:114]: Boolean had the same name as function.

File: ninja.py [hunk 2]
```diff
@@ -441,15 +442,15 @@ def clean(inputSeqsFile, filteredSeqsFile, seqsDBFile, alignmentsFile, parseLogF
...
 
-# Runs ninja, bowtie2 and then processes output. All files output in specified output folder. 
+# Runs ninja, bowtie2 and then processes output. All files output in specified output folder.
 # User must specify ninja's directory as an environment variable named 'NINJA_DIR'
 def main(inputSeqsFile, folder, database, trim, RC, similarity, threads,
         mode, denoising, verboseBool, print_only, stdout, full_output, retain_intermediates):
 
     # Gets ninja's directory relative to current working directory
-    ninjaDirectory = os.path.relpath(os.path.dirname(os.path.realpath(__file__)), os.getcwd()).replace("/", "\\") 
+    ninjaDirectory = os.path.relpath(os.path.dirname(os.path.realpath(__file__)), os.getcwd())
```
Comment (bhillmann) [inline ninja.py:462]: These directory manipulations aren't necessary to my knowledge.

File: ninja.py [hunk 3]
```diff
@@ -248,14 +248,15 @@ def ninja_filter(inputSeqsFile, filteredSeqsFile, seqsDBFile, trim, RC, denoisin
...
     try:
         global shellBool
-        cmd = ninjaFilterFile + ' ' + inputSeqsFile + ' ' + filteredSeqsFile + ' ' + seqsDBFile + ' ' + argTrim + \
-              ' ' + argRC + ' ' + argDenoising
+        # cmd = ninjaFilterFile + ' ' + inputSeqsFile + ' ' + filteredSeqsFile + ' ' + seqsDBFile + ' ' + argTrim + \
+            #   ' ' + argRC + ' ' + argDenoising
+        cmd = [ninjaFilterFile, inputSeqsFile, filteredSeqsFile, seqsDBFile, argTrim, argRC, argDenoising]
         print cmd
-        subprocess.check_call(cmd, shell = shellBool, stdout = sys.stdout)
+        subprocess.check_call(cmd, stdout = sys.stdout)
```
Comment (bhillmann) [inline ninja.py:255]: This call should be cross platform. The use of shell should never happen as it is a security risk. This also leads to the bug with spaces in directory paths.

File: ninja.py [hunk 4]
```diff
@@ -248,14 +248,15 @@ def ninja_filter(inputSeqsFile, filteredSeqsFile, seqsDBFile, trim, RC, denoisin
     cmd = ""
     try:
         global shellBool
-        cmd = ninjaFilterFile + ' ' + inputSeqsFile + ' ' + filteredSeqsFile + ' ' + seqsDBFile + ' ' + argTrim + \
-              ' ' + argRC + ' ' + argDenoising
+        # cmd = ninjaFilterFile + ' ' + inputSeqsFile + ' ' + filteredSeqsFile + ' ' + seqsDBFile + ' ' + argTrim + \
+            #   ' ' + argRC + ' ' + argDenoising
+        cmd = [ninjaFilterFile, inputSeqsFile, filteredSeqsFile, seqsDBFile, argTrim, argRC, argDenoising]
```
Comment (bhillmann) [inline ninja.py:253]: Commands should be in a list.

**Comment by bhillmann** [comment]
I want to resolve #2 where the wrapper doesn't support filepaths with spaces in it.

Should I or should I not remove the ninja_output directory? We might want to set up a git lfs server for this and other lab large files. It could be for the lab on S3 https://github.com/meltingice/git-lfs-s3

**Comment by danknights** [comment]
We need to keep the option of having an output directory . Thanks.

Good idea about s3, let's wait till we really need it though.

Dan

## 

Dan Knights, PhD
Assistant Professor
BioTechnology Institute, Computer Science and Engineering
University of Minnesota
http://www.knightslab.org
cell: 510-900-9015 (dials backwards)
alt: MEMORIZE-PI (636-674-9374)
lab: 612-5-MICROB (612-564-2762)

On Tue, Nov 24, 2015 at 3:06 PM, Benjamin Hillmann <notifications@github.com

**Comment by bhillmann** [comment]
No, I mean having the ninja_output directory as a part of the repository being stored in github. It also has the databases directory which take a long time to pull. Those are the directories that we could store else where such as in lfs.

**Comment by danknights** [comment]
hmm.. was ninja_output added to the repo by mistake? I can't see a reason
why it is there.

Dan

## 

Dan Knights, PhD
Assistant Professor
BioTechnology Institute, Computer Science and Engineering
University of Minnesota
http://www.knightslab.org
cell: 510-900-9015 (dials backwards)
alt: MEMORIZE-PI (636-674-9374)
lab: 612-5-MICROB (612-564-2762)

On Tue, Nov 24, 2015 at 3:29 PM, Benjamin Hillmann <notifications@github.com

**Comment by bhillmann** [comment]
If we aren't hitting a repository cap in Github though it seems like everything is OK for now. Disk quota is 1GB on Github. I'm not sure if it is there on purpose.

**Comment by danknights** [comment]
Yes. I just removed ninja_output from the repo though since I think it was
just an error.

Dan

## 

Dan Knights, PhD
Assistant Professor
BioTechnology Institute, Computer Science and Engineering
University of Minnesota
http://www.knightslab.org
cell: 510-900-9015 (dials backwards)
alt: MEMORIZE-PI (636-674-9374)
lab: 612-5-MICROB (612-564-2762)

On Tue, Nov 24, 2015 at 3:36 PM, Benjamin Hillmann <notifications@github.com

**Pull request closed without merging**